src/main/java/mat/client/Mat.java

```diff
@@ -649,26 +647,19 @@ private void harpOktaLogout(String redirectTo) {
             @Override
             public void onFailure(Throwable throwable) {
                 logger.log(Level.SEVERE, "HarpService::getHarpUrl -> onFailure " + throwable.getMessage(), throwable);
-                removeOktaTokens();
             }
 
             @Override
             public void onSuccess(String harpUrl) {
                 logger.log(Level.INFO, "HarpService::getHarpUrl -> onSuccess");
-                harpLogout(harpUrl);
                 MatContext.get().getSynchronizationDelegate().setLogOffFlag();
-                MatContext.get().handleSignOut("SIGN_OUT_EVENT", redirectTo);
-                removeOktaTokens();
+                MatContext.get().handleSignOut("SIGN_OUT_EVENT", null);
+                harpLogout(harpUrl);
+                redirectToLogin();
             }
         });
     }
 
-    private void removeOktaTokens() {
-        logger.log(Level.INFO, "removeOktaTokens");
```

Comment:
Interesting..

Reply:
Removing the tokens didn't seem to have any affect on logout functionality. I've added the method back to avoid leaving the tokens in memory as a precaution.